drivers/video/video_sw_generator.c

```diff
@@ -56,14 +56,14 @@ static int video_sw_generator_stream_start(const struct device *dev)
 {
 	struct video_sw_generator_data *data = dev->data;
 
-	return k_delayed_work_submit(&data->buf_work, K_MSEC(33));
+	return k_work_schedule(&data->buf_work, K_MSEC(33));
 }
 
 static int video_sw_generator_stream_stop(const struct device *dev)
 {
 	struct video_sw_generator_data *data = dev->data;
 
-	k_delayed_work_cancel(&data->buf_work);
+	k_work_cancel_delayable_sync(&data->buf_work);
```

Comment:
Changing to `_sync` doesn't solve the problem: a cancel can fail, and if it does you have a race condition.

Reply:
Re-reading my own documentation: that's not true; the work item will be idle when this completes.  Note that `stop()` can no longer be invoked from an ISR, though.

Reply:
> Re-reading my own documentation: that's not true; the work item will be idle when this completes.

Ok, good, that was my understanding ;-)

> Note that `stop()` can no longer be invoked from an ISR, though.

Yes, that's fine, both video stream start() and stop() should be (and are in current usages) executed in process context.

Regards,
Loic